packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -43,6 +48,122 @@ export function layerReducer(
         });
       });
     }
+    case 'groupLayer': {
+      const [, id, name] = action;
+
+      const ids = typeof id === 'string' ? [id] : id;
+
+      const page = getCurrentPage(state);
+      const pageIndex = getCurrentPageIndex(state);
+      const indexPaths = Layers.findAllIndexPaths(page, (layer) =>
+        ids.includes(layer.do_objectID),
+      );
+
+      const reversed = [...indexPaths.reverse()];
+      const selectedLayers: (Layers.ChildLayer | Sketch.Group)[] = [];
+
+      const addLayers = (layer: Sketch.AnyLayer) => {
```

Comment:
I was actually thinking I could do this add this code before the produce, since I already have the layers I'll add. 

```
const layers = getSelectedLayersExcludingDescendants(state);
const selectedLayers: Layers.ChildLayer[] = layers.flatMap((layer) =>
        !Layers.isParentLayer(layer) || Layers.isGroup(layer) ? [layer] : [],
      );
```

Reply:
I placed it outside because I remember you mentioned that inside the producer it can be more costly

Reply:
I think the outside approach could be good, but in that case, we'll want to move all logic related to those layers outside also, including the part where we clone them via the nested `produce`. Defining an object/array variable outside of `produce` and modifying it within `produce` can also cause problems

Reply:
I didn't understand 😅 What should I change? place it outside ? or inside?

Reply:
Because if I place it outside I wouldn't have the code that deletes the elements in the parent node

Reply:
Ok, let's go with inside then. It's more important that everything is in one place